src/main/java/com/redhat/devtools/intellij/telemetry/core/service/segment/SegmentBroker.java

```diff
@@ -0,0 +1,201 @@
+        properties.put(PROP_OS_NAME, environment.getPlatform().getName());
+        properties.put(PROP_OS_VERSION, environment.getPlatform().getVersion());
+        properties.put(PROP_TIMEZONE, environment.getTimezone());
+        return properties;
+    }
+
+    private MessageBuilder toMessage(TrackMessage.Builder builder, TelemetryEvent event, Map<String, Object> context) {
+        return builder
+                .anonymousId(anonymousId)
+                .properties(addIdentifyEnvironment(event.getProperties()))
+                .context(context);
+    }
+
+    private Map<String, ?> addIdentifyEnvironment(Map<String, String> properties) {
+        Application application = environment.getApplication();
+        properties.put(PROP_APP_NAME, application.getName());
+        properties.put(PROP_APP_VERSION, application.getVersion());
+        application.getProperties().forEach(
+                appProperty -> properties.put(appProperty.getKey(), String.valueOf(appProperty.getValue())));
+        properties.put(PROP_EXTENSION_NAME, environment.getPlugin().getName());
+        properties.put(PROP_EXTENSION_VERSION, environment.getPlugin().getVersion());
+        return properties;
+    }
+
+    private MessageBuilder toMessage(PageMessage.Builder builder, TelemetryEvent event, Map<String, Object> context) {
+        return builder
+                .anonymousId(anonymousId)
+                .properties(event.getProperties())
+                .context(context);
+    }
+
+    private Map<String, Object> createContext(Environment environment) {
+        return new MapBuilder()
+                .mapPair(PROP_APP)
+                    .pair(PROP_NAME, environment.getApplication().getName())
+                    .pair(PROP_VERSION, environment.getApplication().getVersion())
+                    .pairs(environment.getApplication().getProperties())
+                    .build()
+                .pair(PROP_IP, VALUE_NULL_IP)
+                .pair(PROP_LOCALE, environment.getLocale())
+                .mapPair(PROP_LOCATION)
+                    .pair(PROP_COUNTRY, environment.getCountry())
+                    .build()
+                .mapPair(PROP_OS)
+                    .pair(PROP_NAME, environment.getPlatform().getName())
+                    .pair(PROP_VERSION, environment.getPlatform().getVersion())
+                    .build()
+                .pair(PROP_TIMEZONE, environment.getTimezone())
+                .build();
+    }
+
+    public void dispose() {
+        analytics.get().flush();
+        analytics.get().shutdown();
+    }
+
+    private Analytics createAnalytics(String writeKey) {
+
+        if (writeKey == null) {
+            LOGGER.warn("Could not create Segment Analytics instance, missing writeKey.");
+            return null;
+        }
+        LOGGER.debug("Creating Segment Analytics instance using " + writeKey + " writeKey.");
+        return Analytics.builder(writeKey)
+                .flushQueueSize(1)
```

Comment:
Is 1 the correct value ?

Reply:
vscode-redhat-telemetry had this initially set to 1 so that every message would be sent immediately. Now that they are sure about their reporting they increased the value to 10. Should we really increase it right now?

Reply:
increased to 10 like in vscode-redhat-telemetry